validation/src/main/java/io/serverlessworkflow/validation/StatesValidator.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright 2022-Present The Serverless Workflow Specification Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.serverlessworkflow.validation;
+
+import io.serverlessworkflow.api.Workflow;
+import io.serverlessworkflow.api.defaultdef.DefaultConditionDefinition;
+import io.serverlessworkflow.api.interfaces.State;
+import io.serverlessworkflow.api.states.SwitchState;
+import io.serverlessworkflow.api.timeouts.TimeoutsDefinition;
+import io.serverlessworkflow.api.validation.ValidationError;
+import java.time.Duration;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+final class StatesValidator {
+
+  private static final String NON_NEGATIVE_DURATION_MUST_BE_PROVIDED =
+      "When configured, it must be set with a greater than zero ISO 8601 time duration. For example PT30S.";
+
+  static final String INVALID_EVENT_TIMEOUT_ERROR =
+      "An invalid \"eventTimeout\": \"%s\" configuration was provided for the serverless workflow: \"%s\". "
+          + NON_NEGATIVE_DURATION_MUST_BE_PROVIDED;
+
+  static final String EVENT_TIMEOUT_REQUIRED_ERROR =
+      "The \"eventTimeout\" configuration is required for the \"eventConditions\" based switch state \"%s\" that belongs to the serverless workflow: \"%s\".";
+
+  static final String DATA_CONDITIONS_AND_EVENT_CONDITIONS_FOUND_ERROR =
+      "DataConditions and eventConditions where found at the same time for the switch state \"%s\".";
+
+  static final String NEXT_STATE_NOT_FOUND_FOR_DEFAULT_CONDITION_ERROR =
+      "The \"nextState\" : \"%s\" configured for the \"defaultCondition\" transition in the switch state \"%s\", was not found in the serverless workflow: \"%s\".";
+
+  static final String NEXT_STATE_REQUIRED_FOR_DEFAULT_CONDITION_ERROR =
+      "The \"nextState\" is required for the \"defaultCondition\" transition in the switch state \"%s\" that belongs to the serverless workflow: \"%s\".";
+
+  private StatesValidator() {}
+
+  static List<ValidationError> validateStates(Workflow workflow) {
+    List<State> states = workflow.getStates();
+
+    if (states != null && !states.isEmpty()) {
+      List<ValidationError> errors = new ArrayList<>();
+      for (State state : states) {
+        if (state instanceof SwitchState) {
+          errors.addAll(validateSwitchState(workflow, (SwitchState) state));
```

Comment:
@hbelmiro I think a more OO approach (avoiding instanceof) will be to define a Validator interface 
So the code will look like
for (State state: states) { validators.get(state.getType()).validate(workflow,state))}

Reply:
and then the code will be spread into different state validator rather than having a huge StatesValidator

Reply:
Good idea. Thanks @fjtirado.

Reply:
I would still need to cast `state` to `SwitchState` to have access to its methods.